core/src/test/java/org/ehcache/EhcacheBasicRemoveAllEventsTest.java

```diff
@@ -0,0 +1,284 @@
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+/**
+ * @author
+ */
+public class EhcacheBasicRemoveAllEventsTest extends EhcacheBasicCrudBase {
+
+    @Mock
+    protected CacheEventListener testCacheEventListener;
+
+    protected CacheEventNotificationService cacheEventNotificationService;
+
+    protected IsRemoved isRemoved = new IsRemoved();
+
+    @Mock
+    protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the {@code Set} argument to the
+     * {@link org.ehcache.spi.cache.Store#bulkCompute(java.util.Set, org.ehcache.function.Function, org.ehcache.function.NullaryFunction)
+     *    Store.bulkCompute(Set, Function, NullaryFunction} method.
+     */
+    @Captor
+    private ArgumentCaptor<Set<String>> bulkComputeSetCaptor;
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the
+     * {@link org.ehcache.exceptions.BulkCacheWritingException BulkCacheWritingException}
+     * provided to the
+     * {@link org.ehcache.resilience.ResilienceStrategy#removeAllFailure(Iterable, org.ehcache.exceptions.CacheAccessException, org.ehcache.exceptions.BulkCacheWritingException)}
+     *    ResilienceStrategy.putAllFailure(Iterable, CacheAccessException, BulkCacheWritingException)} method.
+     */
+    @Captor
+    private ArgumentCaptor<BulkCacheWritingException> bulkExceptionCaptor;
+
+    @Test
+    public void testRemoveAllNull() throws Exception {
+        final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+        final FakeStore fakeStore = new FakeStore(originalStoreContent);
+        this.store = spy(fakeStore);
+
+        final Ehcache<String, String> ehcache = this.getEhcache(null);
+        
+        try {
+            ehcache.removeAll(null);
+            fail();
+        } catch (NullPointerException e) {
+            // Expected
+        }
+        verify(testCacheEventListener,never()).onEvent(any(CacheEvent.class));
+
+    }
+
+    /**
+     * Tests {@link Ehcache#removeAll(Set)} for
+     * <ul>
+     *    <li>non-empty request set</li>
+     *    <li>populated {@code Store} - some keys overlap request</li>
+     *    <li>no {@code CacheLoaderWriter}</li>
+     * </ul>
+     */
+    @Test
+    public void testRemoveAll() throws Exception {
+        final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+        final FakeStore fakeStore = new FakeStore(originalStoreContent);
+        this.store = spy(fakeStore);
+
+        final Ehcache<String, String> ehcache = this.getEhcache(null);
+        
+        final Set<String> contentUpdates = fanIn(KEY_SET_A, KEY_SET_C);
+        ehcache.removeAll(contentUpdates);
+        verify(testCacheEventListener,times(12)).onEvent(argThat(isRemoved));
+    }
+
```

Comment:
Same comment about adding test cases for exceptions. This apply too to the rest of the PR